Tests/ClientTests/Frontend/Home/HistoryHighlights/HistoryHighlightsDataAdaptorTests.swift

```diff
@@ -0,0 +1,79 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+@testable import Client
+import XCTest
+import MozillaAppServices
+
+class HistoryHighlightsDataAdaptorTests: XCTestCase {
+
+    var subject: HistoryHighlightsDataAdaptor!
+    var historyManager: MockHistoryHighlightsManager!
+    var notificationCenter: MockNotificationCenter!
+    var delegate: MockHistoryHighlightsDelegate!
+
+    override func setUp() {
+        super.setUp()
+
+        historyManager = MockHistoryHighlightsManager()
+        notificationCenter = MockNotificationCenter()
+        delegate = MockHistoryHighlightsDelegate()
+
+        let subject = HistoryHighlightsDataAdaptorImplementation(
+            historyManager: historyManager,
+            profile: MockProfile(),
+            tabManager: MockTabManager(),
+            notificationCenter: notificationCenter)
+        subject.delegate = delegate
+        notificationCenter.notifiableListener = subject
+        self.subject = subject
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        subject = nil
+        historyManager = nil
+        notificationCenter = nil
+    }
```

Comment:
Should be nullified too?
```suggestion
        notificationCenter = nil
        delegate = nil
    }
```